iwyu_tool.py

```diff
@@ -91,9 +91,24 @@ def clang_formatter(output):
 }
 
 
-IWYU_EXECUTABLE = 'include-what-you-use'
-if sys.platform.startswith('win'):
-    IWYU_EXECUTABLE += '.exe'
+def find_include_what_you_use():
+    """ Find IWYU executable and return its full pathname. """
+    executable_name = 'include-what-you-use'
+    if sys.platform.startswith('win'):
+        executable_name += '.exe'
+
+    search_path = [os.path.dirname(__file__)]
+    search_path += os.environ.get('PATH', '').split(os.pathsep)
```

Comment:
Nice, I didn't know this existed! We can't assume Python 3, though, and I still want to add `dirname(__file__)` as an additional search path entry.

Reply:
Maybe a TODO comment then, for when Python 2 is dropped?

Reply:
I still don't see a way to convince `shutil.which` to look at additional paths beyond `PATH`, so I would have to modify `PATH` before calling it, which is always a mess because of the str/list/pathsep divide.

But maybe I can trade that complexity for the `.exe` normalization. I'll add a TODO.

Reply:
Done!